src/utils/chain/common.ts

```
@@ -1,7 +1,26 @@
+const CACHE_DURATION = 30 * 1000; // 30 seconds
+let cachedPrice: number | null = null;
+let lastFetchTime = 0;
+
 export async function getPriceCoinGecko(token: string, currency: string): Promise<number> {
+    const now = Date.now();
+
+    // Return cached value if within 30 seconds
+    if (cachedPrice !== null && now - lastFetchTime < CACHE_DURATION) {
+        return cachedPrice;
+    }
+
+    // Fetch new price from CoinGecko
     const res = await fetch(
         `https://api.coingecko.com/api/v3/simple/price?ids=${token}&vs_currencies=${currency}`
     ).then((res) => res.json());
 
-    return res?.ethereum?.usd;
+    const price = res?.ethereum?.usd;
```

better type safety to throw an error if the object is not expected, instead of using nullifier accessors